src/commands/appdistribution-distribute.ts

```diff
@@ -210,56 +224,78 @@ export const command = new Command("appdistribution:distribute <release-binary-f
     await requests.distribute(releaseName, testers, groups);
 
     // Run automated tests
-    if (testDevices?.length) {
-      utils.logBullet("starting automated tests (note: this feature is in beta)");
-      const releaseTest = await requests.createReleaseTest(
-        releaseName,
-        testDevices,
-        loginCredential,
-      );
-      utils.logSuccess(`Release test created successfully`);
+    if (testDevices.length) {
+      utils.logBullet("starting automated test (note: this feature is in beta)");
+      const releaseTestPromises: Promise<ReleaseTest>[] = [];
+      if (!testCases.length) {
+        // fallback to basic automated test
+        releaseTestPromises.push(
+          requests.createReleaseTest(releaseName, testDevices, loginCredential),
+        );
+      } else {
+        for (const testCaseId of testCases) {
+          releaseTestPromises.push(
+            requests.createReleaseTest(
+              releaseName,
+              testDevices,
+              loginCredential,
+              `${appName}/testCases/${testCaseId}`,
+            ),
+          );
+        }
+      }
+      const releaseTests = await Promise.all(releaseTestPromises);
+      utils.logSuccess(`${releaseTests.length} release test(s) started successfully`);
```

Comment:
"[all] started successfully" -- what if some could not be started due to out-of-quota or other reasons?
There doesn't seem to be any handling of that; this phrasing might be deceptive.

Reply:
From the perspective of the App Distribution API, the "release test" has been started as soon as `createReleaseTest()` returns successfully.

If the test matrix that's used "under the hood" fails for some reason (incl. out-of-quota), this will be caught below (line 279) - unless `--test-non-blocking` is specified. In that case the customer has to check the Firebase Console.

I understand that the TestLab API distinguishes the setup phase from running the actual test, but App Distribution doesn't.